subsys/net/lib/websocket/websocket.c

```diff
@@ -409,8 +409,6 @@ int websocket_disconnect(int ws_sock)
 
 	NET_DBG("[%p] Disconnecting", ctx);
 
-	(void)close(ctx->sock);
```

Comment:
We would leak the fd if the user does not call `close()` but only `websocket_disconnect()`, so some more checks are needed here.